services/catarse/spec/factories/balance_transfers_factories.rb

```diff
@@ -6,5 +6,37 @@
     association :user, :with_bank_account
 
     amount { 50 }
+
+    after :create do |balance_transfer|
+      create(:balance_transfer_transition,
+        to_state: 'pending',
+        balance_transfer: balance_transfer,
+        most_recent: false
+      )
+    end
+
+    factory :processing_balance_transfer do
+      association :project
+      association :user, :with_bank_account
+
+      after :create do |balance_transfer|
+        create(:balance_transfer_transition_authorized,
+          to_state: 'processing',
+          balance_transfer: balance_transfer
+        )
+      end
+    end
+  end
```

Comment:
**Sugestão**
Aqui tu pode fazer assim pra evitar repetição de código e a criação de mais uma factory.

```suggestion
    transient do
      transition_state { 'pending' }
    end
    
    after :create do |balance_transfer, evaluator|
      create(:balance_transfer_transition,
        to_state: evaluator.transition_state,
        balance_transfer: balance_transfer,
        most_recent: false
      )
    end
  end
```

E quando for utilizar essa factory, faz assim:
```ruby
create(:balance_transfer, transition_state: 'processing')
```

Reply:
Aplicado. Realmente ficou muito melhor assim.